mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/param/DefaultEncoding.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.annotation.param;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks that the annotated field or argument should be populated with Mule's default encoding value if no value was given. This
```

Comment:
This is not how this should behave. This shouldn't work with `@Optional`. This should carry its own semantic meaning. Parameters annotated with this should automatically be optional and should automatically default to the runtime encoding if the user didn't provide a value.